grammarinator/generate.py

```diff
@@ -130,8 +134,11 @@ def create_new_test(self, index):
             self.population.add_tree(tree_fn)
             tree.save(tree_fn)
 
-        with codecs.open(test_fn, 'w', self.encoding) as f:
-            f.write(self.serializer(tree.root))
+        if test_fn:
+            with codecs.open(test_fn, 'w', self.encoding) as f:
+                f.write(self.serializer(tree.root))
+        else:
+            print(self.serializer(tree.root))
```

Comment:
If grammarinator runs in multiprocess model, how will be ensured that the result of various threads won't be interleaved on the stdout? Maybe a lock should be shared between the threads and enable to print only for the holder of the lock.